adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/FeatureRDDFunctions.scala

```diff
@@ -133,11 +132,167 @@ class FeatureRDDFunctions(featureRDD: RDD[Feature]) extends Serializable with Lo
+      val start = feature.getStart
+      val end = feature.getEnd
+      val name = Features.nameOf(feature)
+      val score = Option(feature.getScore).getOrElse(".")
+      val strand = Features.toString(feature.getStrand)
+
+      if (!feature.getAttributes.containsKey("thickStart")) {
+        // write BED6 format
+        List(chrom, start, end, name, score, strand).mkString("\t")
+      } else {
+        // write BED12 format
+        val thickStart = feature.getAttributes.getOrElse("thickStart", ".")
+        val thickEnd = feature.getAttributes.getOrElse("thickEnd", ".")
+        val itemRgb = feature.getAttributes.getOrElse("itemRgb", ".")
+        val blockCount = feature.getAttributes.getOrElse("blockCount", ".")
+        val blockSizes = feature.getAttributes.getOrElse("blockSizes", ".")
+        val blockStarts = feature.getAttributes.getOrElse("blockStarts", ".")
+        List(chrom, start, end, name, score, strand, thickStart, thickEnd, itemRgb, blockCount, blockSizes, blockStarts).mkString("\t")
+      }
+    }
+    featureRDD.map(toBed).saveAsTextFile(fileName)
+  }
+
+  def saveAsIntervalList(fileName: String) = {
+    def escape(entry: (Any, Any)): String = {
+      entry._1 + "|" + entry._2
+    }
+
+    def toInterval(feature: Feature): String = {
+      val sequenceName = feature.getContigName
+      val start = feature.getStart + 1 // IntervalList ranges are 1-based
+      val end = feature.getEnd // IntervalList ranges are closed
+      val strand = Features.toString(feature.getStrand)
+      val intervalName = Features.nameOf(feature)
+      List(sequenceName, start, end, strand, intervalName).mkString("\t")
+    }
+    // todo:  SAM style header
+    featureRDD.map(toInterval).saveAsTextFile(fileName)
+  }
+
+  def saveAsNarrowPeak(fileName: String) {
+    def toPeak(feature: Feature): String = {
+      val chrom = feature.getContigName
+      val start = feature.getStart
+      val end = feature.getEnd
+      val name = Features.nameOf(feature)
+      val score = Option(feature.getScore).getOrElse(".")
+      val strand = Features.toString(feature.getStrand)
+      val signalValue = feature.getAttributes.getOrElse("signalValue", "0")
+      val pValue = feature.getAttributes.getOrElse("pValue", "-1")
+      val qValue = feature.getAttributes.getOrElse("qValue", "-1")
+      val peak = feature.getAttributes.getOrElse("peak", "-1")
+      List(chrom, start, end, name, score, strand, signalValue, pValue, qValue, peak).mkString("\t")
+    }
+    featureRDD.map(toPeak).saveAsTextFile(fileName)
+  }
+
+  def sortByExon(): RDD[Feature] = {
+    def exon(f: Feature) =
+      (
+        f.getContigName,
+        f.getStart,
+        f.getEnd,
+        f.getFeatureType,
+        // try exon_id first
+        Option(f.getExonId).getOrElse("~"),
+        // then try transcript_id and exon_number
+        Option(f.getTranscriptId).getOrElse("~"),
+        f.getAttributes.getOrElse("exon_number", Int.MaxValue.toString).toInt
```

Comment:
nulls don't work in scala Ordering, and this line doesn't compile or handle nulls